### ISSUE TITLE
How to make translation for filter texts in syncfusion datagrid 

### ISSUE THREAD

**Issue by omar0141** [syncfusion/flutter-widgets#1113]
I am trying to make translation for filter texts in syncfusion datagrid and i used a localization but still didn't work with me

**Comment by omar0141** [comment]
![Screenshot_20230305_093136_com syncfusion flutter examples](https://user-images.githubusercontent.com/75172714/222947843-cc393c7b-e90e-4b95-a3cb-ea7a525e1b86.jpg)

**Comment by Tamilarasan-Paranthaman** [comment]
Hi @omar0141,

Based on the information you provided, we were unable to reproduce the reported issue on our end. It appears to be working correctly. However, please ensure that you have imported the [syncfusion_localizations](https://pub.dev/packages/syncfusion_localizations) package into your application. For your reference, I have attached a simple sample along with the code snippet below.

```dart
import 'package:flutter/material.dart';
import 'package:syncfusion_flutter_datagrid/datagrid.dart';
import 'package:syncfusion_localizations/syncfusion_localizations.dart';
import 'package:flutter_localizations/flutter_localizations.dart';

void main() {
  runApp(MaterialApp(
      localizationsDelegates: const [
        GlobalMaterialLocalizations.delegate,
        GlobalCupertinoLocalizations.delegate,
        GlobalWidgetsLocalizations.delegate,
        SfGlobalLocalizations.delegate
      ],
      supportedLocales: const [
        Locale('zh'),
        Locale('ar'),
        Locale('ja'),
      ],
      locale: const Locale('zh'),
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const Grid()));
}

```
![image](https://user-images.githubusercontent.com/93904422/223046379-b3cebd80-7dcd-420c-9238-4f1fa6c0dddd.png)

Regards,
Tamilarasan

**Comment by omar0141** [comment]
Thank you it worked for me but i found a bug in it i guess.
This text in red square is not showing right
![Screenshot 2023-03-06 211759](https://user-images.githubusercontent.com/75172714/223209250-932053d1-470e-4300-991d-d90b0e5ecc0e.png)

**Comment by omar0141** [comment]
and there are some words it doesn't have the right mean
![Screenshot 2023-03-06 212422](https://user-images.githubusercontent.com/75172714/223210526-ee347027-06d8-4eed-8db5-da52cac1c223.png)
this isn't "يلغى" it is "إلغاء" mean cancel in English.

**Comment by omar0141** [comment]
and this isn't "مرشح واضح" it is "مسح المرشح" 
"clear" has two means in arabic on it is "واضح" and the other is "مسح" 
but in this sentence it is mean "مسح" not "واضح"
![Screenshot 2023-03-06 212743](https://user-images.githubusercontent.com/75172714/223211909-1cc7fe63-ba62-4d87-b734-0518826b1fa0.png)

**Comment by Tamilarasan-Paranthaman** [comment]
@omar0141,

**Regarding query: This text in red square is not [showing](https://github.com/syncfusion/flutter-widgets/issues/1113#issuecomment-1456814246) right.**

We believe that you are experiencing an issue on the web platform, and we have been able to reproduce the problem. However, the application works fine on the Windows platform. We tested a simple sample and encountered the same issue with Arabic text on the web platform. We have reported this issue to the framework, and you can find more details on the link below. 

GitHub: https://github.com/flutter/flutter/issues/122284 

**Regarding query:  and there are some words it doesn't have the right mean.**

Thank you for bringing this issue to our attention. We plan to address it in our upcoming 2023 Volume 1 release, which is scheduled for the end of March 2023. We will inform you as soon as the update is rolled out. We appreciate your patience and understanding in the meantime.

**Comment by omar0141** [comment]
Good evening,
Thanks for your all efforts, 
i took those screenahots from windows application not web framework

**Comment by Tamilarasan-Paranthaman** [comment]
@omar0141,

**Regarding Query: I took those screenshots from windows application not web framework.**

We have tested this issue on the **`Windows`** platform and have not faced the [mentioned ](https://github.com/syncfusion/flutter-widgets/issues/1113#issuecomment-1456814246)problem, it works properly. However, we have encountered this issue only on the **`Web`** platform. Our tests were conducted using the latest Flutter stable SDK version **3.7.7** and DataGrid version **20.4.53**. For your reference, we have attached an output image of the issue as it appears on the Windows platform.


![image](https://user-images.githubusercontent.com/93904422/224665319-73e95a44-f4d4-460f-88f0-f9bfef02b395.png)


Furthermore, we have received confirmation from the framework team that there is a difference between the Web and Windows platforms. To further investigate this issue, could you please provide us with details regarding the Flutter channel and version you are using, as well as the DataGrid configuration?

**Comment by omar0141** [comment]
Okay Thank you, i will try it again

**Comment by Tamilarasan-Paranthaman** [comment]
@omar0141,

The fix for the reported "**localization text does not correct**" [issue](https://github.com/syncfusion/flutter-widgets/issues/1113#issuecomment-1456833184) has been included in the latest DataGrid version **21.1.37**. Update your DataGrid to get the issue resolved.

Regards,
Tamilarasan

**Comment by ashok-kuvaraja** [comment]
Hi @omar0141,

We suspect that the reported problem has been resolved at your end. Please let us know if you have any further queries on this. We are happy to help you.

Regards, 
Ashok K
